public/app/features/dashboard-scene/serialization/transformToV1TypesUtils.ts

```diff
@@ -0,0 +1,131 @@
+import { FieldConfigSource as FieldConfigSourceV1, SpecialValueMatch as SpecialValueMatchV1 } from '@grafana/data';
+import {
+  VariableHide as VariableHideV1,
+  VariableRefresh as VariableRefreshV1,
+  VariableSort as VariableSortV1,
+  DashboardCursorSync as DashboardCursorSyncV1,
+  defaultDashboardCursorSync,
+} from '@grafana/schema';
+import {
+  DashboardCursorSync,
+  MappingType,
+  VariableHide,
+  VariableRefresh,
+  VariableSort,
+  FieldConfigSource,
+  SpecialValueMatch,
+} from '@grafana/schema/dist/esm/schema/dashboard/v2alpha0/dashboard.gen';
+
+export function transformVariableRefreshToEnumV1(refresh?: VariableRefresh): VariableRefreshV1 {
+  switch (refresh) {
+    case VariableRefresh.Never:
+      return VariableRefreshV1.never;
+    case VariableRefresh.OnDashboardLoad:
+      return VariableRefreshV1.onDashboardLoad;
+    case VariableRefresh.OnTimeRangeChanged:
+      return VariableRefreshV1.onTimeRangeChanged;
+    default:
+      return VariableRefreshV1.never;
+  }
+}
+
+export function transformVariableHideToEnumV1(hide?: VariableHide): VariableHideV1 {
+  switch (hide) {
+    case VariableHide.DontHide:
+      return VariableHideV1.dontHide;
+    case VariableHide.HideLabel:
+      return VariableHideV1.hideLabel;
+    case VariableHide.HideVariable:
+      return VariableHideV1.hideVariable;
+    default:
+      return VariableHideV1.dontHide;
+  }
+}
+
+export function transformSortVariableToEnumV1(sort?: VariableSort): VariableSortV1 {
+  switch (sort) {
+    case VariableSort.Disabled:
+      return VariableSortV1.disabled;
+    case VariableSort.NumericalAsc:
+      return VariableSortV1.numericalAsc;
+    case VariableSort.NumericalDesc:
+      return VariableSortV1.numericalDesc;
+    case VariableSort.AlphabeticalAsc:
+      return VariableSortV1.alphabeticalAsc;
+    case VariableSort.AlphabeticalDesc:
+      return VariableSortV1.alphabeticalDesc;
+    default:
+      return VariableSortV1.disabled;
+  }
+}
+
+export function transformCursorSyncV2ToV1(cursorSync: DashboardCursorSync): DashboardCursorSyncV1 {
+  switch (cursorSync) {
+    case DashboardCursorSync.Crosshair:
+      return DashboardCursorSyncV1.Crosshair;
+    case DashboardCursorSync.Tooltip:
+      return DashboardCursorSyncV1.Tooltip;
+    case DashboardCursorSync.Off:
+      return DashboardCursorSyncV1.Off;
+    default:
+      return defaultDashboardCursorSync;
+  }
+}
+
+function transformSpecialValueMatchToV1(match: SpecialValueMatch): SpecialValueMatchV1 {
```

Comment:
I'd ask @K-Phoen if we have a way to solve this without transformations. If they TS are the same named TS, and it is a TS generation error, we should be able to fix it from Cog :) Let's create an issue for this to remove this transformer. But let merge it for now like this to unblock

Reply:
Yeah, I totally agree we should support this in Cog :)

Reply:
I've added this issue to the epic